internal/controller/management_controller.go

```diff
@@ -772,6 +775,32 @@ func updateComponentsStatus(
 	}
 }
 
+// setReadyCondition updates the Management resource's "Ready" condition based on whether
+// all components are healthy.
+func setReadyCondition(management *kcm.Management) {
+	var failing []string
+	for name, comp := range management.Status.Components {
+		if !comp.Success {
+			failing = append(failing, name)
+		}
+	}
+
+	readyCond := metav1.Condition{
+		Type:               kcm.ReadyCondition,
+		ObservedGeneration: management.Generation,
+		Status:             metav1.ConditionTrue,
+		Reason:             kcm.AllComponentsHealthyReason,
+		Message:            "All components are successfully installed.",
+	}
+	if len(failing) > 0 {
```

Comment:
I suggest sorting the failing components list before displaying them to prevent unnecessary updates to the Management status, as the list keeps shuffling.

Reply:
That's a great suggestion! Added and Thanks.